dev/Repeater/APITests/LayoutTests.cs

```diff
@@ -192,6 +192,33 @@ public void ValidateNonVirtualLayoutDoesNotGetMeasuredForViewportChanges()
             });
         }
 
+        [TestMethod]
+        public void ValidateStackLayoutDoesNotRetainIncorrectMinorWidth()
+        {
+            RunOnUIThread.Execute(() =>
+            {
+                var repeater = new ItemsRepeater();
+                var stackLayout = new StackLayout();
+                repeater.Layout = stackLayout;
+                repeater.ItemsSource = Enumerable.Range(0, 1);
+                repeater.ItemTemplate = (DataTemplate)XamlReader.Load(
+                    @"<DataTemplate  xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'>
+                         <Button Content='{Binding}' Height='100' />
+                    </DataTemplate>");
+                var scrollViewer = new ScrollViewer() {
+                    Content = repeater
+                };
+                scrollViewer.Width = 300;
+                Content = scrollViewer;
+                Content.UpdateLayout();
+
+                repeater.Measure(new Size(600, 100));
+                Content.UpdateLayout();
+
+                Verify.AreEqual(repeater.ActualSize.X, 300);
```

Comment:
UIElement.ActualSize was added in 19h1, it actually a facade to the underlying visual's size. You should be using FrameworkElements.ActualWidth as what you have will crash on earlier OS versions.

Reply:
Good catch.